ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -116,13 +118,13 @@ public class CloudBlobStore implements Store {
     minTtlMillis = TimeUnit.DAYS.toMillis(cloudConfig.vcrMinTtlDays);
     requireEncryption = cloudConfig.vcrRequireEncryption;
     isVcr = cloudConfig.cloudIsVcr;
-    if (isVcr) {
+    if (isVcr && cloudConfig.ambryBackupVersion.equals(CloudConfig.AMBRY_BACKUP_VERSION_1)) {
```

Comment:
There is CloudBlobStoreV2. Are you still using it?
If yes, should the v2 code be there?
Looks like there are many implementation difference between V1 and V2, removed retry, removed cache, removed WriteChannel.

Reply:
I will delete all V2 classes next month. I designed them to avoid these cases and implement separately from V1 but that idea was not accepted early this semester. Right now, I don't have the time to refactor and go back to it. 

I tried to set the max cache size to 0 in configs, but the V1 code is so bad that it silently creates a cache of size 1000.

Reply:
```
  private static final int cacheInitialCapacity = 1000;

```